sql/src/main/java/io/crate/operation/udf/UDFLanguage.java

```diff
@@ -0,0 +1,17 @@
+package io.crate.operation.udf;
+
+import io.crate.metadata.FunctionImplementation;
+
+import javax.script.ScriptException;
+
+
+public interface UDFLanguage {
+
+    public FunctionImplementation createFunctionImplementation(UserDefinedFunctionMetaData metaData) throws ScriptException;
+
+    public void validate(UserDefinedFunctionMetaData metadata);
```

Comment:
should validate return bool for success/failure or throw an exeption?
if it throws, there should be a `throws InvalidFunction` something in the signature